ocaml/libs/tracing/tracing.ml

```diff
@@ -803,19 +804,30 @@ module Export = struct
       |> List.concat_map (fun x -> TracerProvider.get_endpoints x)
       |> List.iter (export_to_endpoint parent span_list)
 
+    let delay = Delay.make ()
+
+    (* Note this signal will flush the spans and terminate the exporter thread *)
+    let signal () = Delay.signal delay
+
     let main () =
       enable_span_garbage_collector () ;
       Thread.create
         (fun () ->
           while true do
             debug "Tracing: Waiting %d seconds before exporting spans"
               (int_of_float !export_interval) ;
-            Thread.delay !export_interval ;
+            if not (Delay.wait delay !export_interval) then (
+              debug "Tracing: we are signaled, export spans now and exit" ;
+              flush_spans () ;
+              raise Thread.Exit
```

Comment:
I'd prefer to change the while loop to exit naturally rather than raising an exception.

Reply:
I would prefer if a separate function was provided for lwt-based executables, which returns a promise (` -> unit Lwt.t`).
But that probably needs a separate library to avoid including Lwt into xapi

Reply:
> exit naturally rather than raising an exception

you mean having a mutable flag and check for that?

Reply:
> But that probably needs a separate library to avoid including Lwt into xapi

This probably needs a larger effort of introudcing Lwt into xapi

Reply:
> you mean having a mutable flag and check for that?

Yes, your last update is what I meant.